internal/ci/ci.go

```diff
@@ -81,3 +94,177 @@ func GetProvider() Provider {
+		ci = CI{
+			Provider:  provider,
+			OriginURL: os.Getenv("CIRCLE_BUILD_URL"),
+			Repo:      os.Getenv("CIRCLE_REPOSITORY_URL"),
+			RefName:   os.Getenv("CIRCLE_BRANCH"),
+			SHA:       os.Getenv("CIRCLE_SHA1"),
+			User:      os.Getenv("CIRCLE_USERNAME"),
+		}
+	}
+	if reflect.DeepEqual(provider, CodeShip) {
+		ci = CI{
+			Provider:  provider,
+			OriginURL: os.Getenv("CI_BUILD_URL"),
+			Repo:      os.Getenv("CI_REPO_NAME"),
+			RefName:   os.Getenv("CI_BRANCH"),
+			SHA:       os.Getenv("CI_COMMIT_ID"),
+		}
+	}
+	if reflect.DeepEqual(provider, Drone) {
+		ci = CI{
+			Provider:  provider,
+			OriginURL: os.Getenv("DRONE_BUILD_LINK"),
+			Repo:      os.Getenv("DRONE_REPO"),
+			RefName:   os.Getenv("DRONE_BRANCH"),
+			SHA:       os.Getenv("DRONE_COMMIT_SHA"),
+		}
+	}
+	if reflect.DeepEqual(provider, GitHub) {
+		ci = CI{
+			Provider:  provider,
+			OriginURL: fmt.Sprintf("%s/%s/actions/runs/%s", os.Getenv("GITHUB_SERVER_URL"), os.Getenv("GITHUB_REPOSITORY"), os.Getenv("GITHUB_RUN_ID")),
+			Repo:      os.Getenv("GITHUB_REPOSITORY"),
+			RefName:   os.Getenv("GITHUB_REF_NAME"),
+			SHA:       os.Getenv("GITHUB_SHA"),
+			User:      os.Getenv("GITHUB_ACTOR"),
+		}
+	}
+	if reflect.DeepEqual(provider, GitLab) {
+		ci = CI{
+			Provider:  provider,
+			OriginURL: os.Getenv("CI_JOB_URL"),
+			Repo:      os.Getenv("CI_PROJECT_PATH"),
+			RefName:   os.Getenv("CI_COMMIT_REF_NAME"),
+			SHA:       os.Getenv("CI_COMMIT_SHA"),
+			User:      os.Getenv("GITLAB_USER_LOGIN"),
+		}
+	}
+
+	if reflect.DeepEqual(provider, Jenkins) {
+		ci = CI{
+			Provider:  provider,
+			OriginURL: os.Getenv("JOB_URL"),
+			Repo:      os.Getenv("GIT_URL"),
+			RefName:   os.Getenv("GIT_BRANCH"),
+			SHA:       os.Getenv("GIT_COMMIT"),
+		}
+	}
+	if reflect.DeepEqual(provider, Semaphore) {
+		ci = CI{
+			Provider:  provider,
+			OriginURL: fmt.Sprintf("%s/workflows/%s?pipeline_id=%s", os.Getenv("SEMAPHORE_ORGANIZATION_URL"), os.Getenv("SEMAPHORE_PROJECT_ID"), os.Getenv("SEMAPHORE_JOB_ID")),
+			Repo:      os.Getenv("SEMAPHORE_GIT_URL"),
+			RefName:   os.Getenv("SEMAPHORE_GIT_BRANCH"),
+			SHA:       os.Getenv("SEMAPHORE_GIT_SHA"),
+		}
+	}
+	if reflect.DeepEqual(provider, Travis) {
+		ci = CI{
+			Provider:  provider,
+			OriginURL: os.Getenv("TRAVIS_BUILD_WEB_URL"),
+			Repo:      os.Getenv("TRAVIS_REPO_SLUG"),
+			RefName:   os.Getenv("TRAVIS_BRANCH"),
+			SHA:       os.Getenv("TRAVIS_COMMIT"),
+		}
+	}
+	if reflect.DeepEqual(provider, TeamCity) {
+		ci = CI{
+			Provider: provider,
+		}
+	}
+
+	return ci
+}
+
+// GetTags returns tag list containing CI info
+func GetTags() []string {
+	provider := GetProvider()
+	if reflect.DeepEqual(provider, None) {
+		return []string{}
+	}
+
+	ci := getCI(provider)
+	return []string{
+		ci.Provider.Name,
+		fmt.Sprintf("%s:%s", "originURL", ci.OriginURL),
+		fmt.Sprintf("%s:%s", "repo", ci.Repo),
+		fmt.Sprintf("%s:%s", "refName", ci.RefName),
+		fmt.Sprintf("%s:%s", "SHA", ci.SHA),
+		fmt.Sprintf("%s:%s", "user", ci.User),
```

Comment:
So some of these can be potentially empty. Is there a point to adding an empty tag with `user:` etc.?

Reply:
updated